tests/test_api.py

```diff
@@ -599,12 +599,11 @@ def setUp(self):
         self.upgrade_deadline = datetime.now() + timedelta(days=1)  # 1 day from now
 
     @freeze_time('2024-01-01')
-    @patch('learning_assistant.api.CourseMode')
-    def test_upgrade_deadline_expired(self, mock_course_mode):
-
-        mock_mode = MagicMock()
-        mock_mode.expiration_datetime.return_value = datetime.now() - timedelta(days=1)  # yesterday
-        mock_course_mode.objects.get.return_value = mock_mode
+    @patch('learning_assistant.api.CourseEnrollment')
+    def test_upgrade_deadline_expired(self, mock_course_enrollment):
+        # mock_enrollment = MagicMock(upgrade_deadline=)
+        mock_course_enrollment.return_value.upgrade_deadline.return_value = datetime.now() - timedelta(days=1)  # yesterday
```

Comment:
this returns a MagicMock object instead of a datetime... not sure why! Spent a bit of time and referenced this: https://stackoverflow.com/questions/38199008/python-returns-magicmock-object-instead-of-return-value

the signature looks like this: `def audit_trial_is_expired(enrollment, audit_trial_data)`

And the call to get the upgrade_deadline is this: upgrade_deadline = enrollment.upgrade_deadline. 

I wonder if it is weird because I'm mocking the course enrollment when it is a param rather than the other similar cases where the mocked object is retrieved via a getter.

Reply:
All resolved! just needed to remove the return value.